src/coreclr/vm/stublink.cpp

```diff
@@ -1620,11 +1623,11 @@ bool StubLinker::EmitUnwindInfo(Stub* pStubRX, Stub* pStubRW, int globalsize, Lo
 
         _ASSERTE(IS_ALIGNED(pCurFunction, sizeof(void*)));
 
-        S_UINT32 sBeginAddress = S_BYTEPTR(pCode) - S_BYTEPTR(pbBaseAddress);
+        S_UINT32 sBeginAddress = S_BYTEPTR(pCode) - S_BYTEPTR(pbBaseAddressRX);
         if (sBeginAddress.IsOverflow())
             COMPlusThrowArithmetic();
 
-        S_UINT32 sTemp = S_BYTEPTR(pUnwindInfo) - S_BYTEPTR(pbBaseAddress);
+        S_UINT32 sTemp = S_BYTEPTR(pUnwindInfo) - S_BYTEPTR(pbBaseAddressRX);
```

Comment:
This `sTemp` uses `pbBaseAddressRX` in this case, but the other `sTemp` instance uses `pbBaseAddressRW`. What is the logic here?

Reply:
> Disclaimer: I don't fully understand the logic here, and unwinding is still not functional yet.

The code was infinitely looping before my fix, and when integer finally overflows, it crashes with an ArithmeticOverflow exception later.

The idea of my fix is that we should only do arithmetic if they comes from the same mapping. It is wrong to compute the difference of two pointers coming from different mappings, and if they come from the same mapping, the difference should be the same.

This instance you spotted is committing that mistake again. I will get it fixed.

Unfortunately, that still didn't fix the unwind, yet.